admin/RemoveEmptyAccounts.pl

```diff
@@ -0,0 +1,189 @@
+  AND NOT EXISTS (SELECT 1
+                    FROM place_tag_raw
+                   WHERE place_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM recording_tag_raw
+                   WHERE recording_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM release_tag_raw
+                   WHERE release_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM release_group_tag_raw
+                   WHERE release_group_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM series_tag_raw
+                   WHERE series_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM work_tag_raw
+                   WHERE work_tag_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM artist_rating_raw
+                   WHERE artist_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM event_rating_raw
+                   WHERE event_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM label_rating_raw
+                   WHERE label_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM place_rating_raw
+                   WHERE place_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM recording_rating_raw
+                   WHERE recording_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM release_group_rating_raw
+                   WHERE release_group_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM work_rating_raw
+                   WHERE work_rating_raw.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                    FROM annotation
+                   WHERE annotation.editor = e.id)
+  AND NOT EXISTS (SELECT 1
+                   FROM editor_collection ec
+                  WHERE ec.editor = e.id
+                    AND ((
+                              (type != 5 OR name != 'Attending')
+                              AND
+                              (type != 6 OR name != 'Maybe attending')
+                         )
+                         OR EXISTS (SELECT 1
+                                      FROM editor_collection_event ece
+                                     WHERE ece.collection = ec.id)))
+             AND (   last_login_date IS NULL
+                  OR last_login_date < NOW() - INTERVAL '12 months')
+  SQL
+
+$c_cursor->sql->begin;
+$c_cursor->sql->do("DECLARE cursor NO SCROLL CURSOR FOR $query");
+
+my $pull = sub {
+    $c_cursor->sql->select("FETCH FORWARD $limit FROM cursor");
+};
+while ($pull->()) {
+    $c->sql->begin;
+    my @row;
+    while (@row = $c_cursor->sql->next_row) {
+      if ($dry_run) {
+          print 'Removing account ' . $row[1] . " (dry run)\n";
+      } else {
+          print 'Removing account ' . $row[1] . "\n";
+          $c->model('Editor')->delete($row[0]);
```

Comment:
Given that these accounts were never actually used, should we change this to allow name reuse by default?

Reply:
That actually seems sensible to me. @Freso, what do you think?

Reply:
Yeah, I’m good with that. 👍